nexus/lib/qmcpack.py

```diff
@@ -41,16 +41,524 @@
+        if (twistnum_input is not None) or (supercell_nkpoints == 1):
+            self.error('''
+                It appears that a single-twist QMC run was attempted using gcta keyword.
+                Currently, this is not supported. Please contact the developers if this is needed.''')
+        #end if
+    #end def check_implementation
+
+    @staticmethod
+    def int_kpoint_weight(float_value, atol=1e-8):
+        '''
+        This function checks if the float k-point weight/norm is close to its integer value. If so, returns the integer value.
+        '''
+        int_value = round(float_value)
+        assert abs(float_value - int_value) < atol, '''
+            The k-point weight or norm ({}) is not close to an integer!
+            There might be a problem with how the weights were stored.
+            Please check the SCF conversion step.
+            '''.format(float_value)
+        return int_value
+    #end def check_kpoint_weight
+
+    def read_eshdf_data(self, filename):
+        '''
+        Read the ESHDF eigenvalues, k-point info and store the data in the GCTA instance as an attribute
+        '''
+        def h5int(i):
+            return array(i,dtype=int)[0]
+        #end def h5int
+        h        = read_hdf(filename,view=True)
+        nkpoints = h5int(h.electrons.number_of_kpoints)
+        nspins   = h5int(h.electrons.number_of_spins)
+        data     = obj()
+        kweights = []
+        for ikpoint in range(nkpoints):
+            kp = h.electrons['kpoint_'+str(ikpoint)]
+            kw = array(kp.weight)[0]
+            kweights.append(kw)
+            for ispin in range(nspins):
+                path = 'electrons/kpoint_{0}/spin_{1}'.format(ikpoint,ispin)
+                spin = h.get_path(path)
+                eigs = convert(array(spin.eigenvalues),'Ha','eV')
+                nstates = h5int(spin.number_of_states)
+                data[ikpoint,ispin] = obj(
+                    eig    = array(eigs),
+                    kpoint = array(kp.reduced_k), # unit (crystal) coordinates for kpoints. The range is [0, 1).
+                    kweight = kw,
+                    )
+            #end for
+        #end for
+        total_kweight = sum(kweights)
+        total_kweight = self.int_kpoint_weight(total_kweight)
+        norm_factor = 1.0 / min(kweights) # Multiplicative factor to get integer weights
+        norm_factor = self.int_kpoint_weight(norm_factor)
+        res = obj(
+            orbfile     = filename,
+            nkpoints    = nkpoints,
+            total_kw    = total_kweight,
+            norm_factor = norm_factor,
+            nspins      = nspins,
+            nstates     = nstates,
+            data        = data,
+            )
+        self.eig_data = res
+    #end def read_eshdf_data
+
+    def unfolded_nelecs(self):
+        '''
+        Returns the number of electrons in the primitive cell
+        '''
+        if self.system.folded_system == None:
```

Comment:
Change to "is None".  Using "==" actually can cause problems for certain types, "is" is safer.